devito/types/sparse.py

```diff
@@ -826,10 +838,21 @@ def _dist_gather(self, data):
         mpitype = MPI._typedict[np.dtype(self.dtype).char]
         comm.Alltoallv([data, rcount, rdisp, mpitype],
                        [gathered, scount, sdisp, mpitype])
-        data = gathered
         # Unpack data values so that they follow the expected storage layout
-        data = np.ascontiguousarray(np.transpose(data, self._dist_reorder_mask))
-        self._data[:] = data[self._dist_gather_mask]
+        gathered = np.ascontiguousarray(np.transpose(gathered, self._dist_reorder_mask))
+        self._data[:] = gathered[self._dist_gather_mask]
+
+        if coords is not None:
+            # Pack (reordered) coordinates so that they can be sent out via an Alltoallv
+            coords = coords + np.array(self.grid.origin_offset, dtype=self.dtype)
+            # Send out the sparse point coordinates
+            sshape, scount, sdisp, _, rcount, rdisp = self._dist_subfunc_alltoall
+            gathered = np.empty(shape=sshape, dtype=self.coordinates.dtype)
+            mpitype = MPI._typedict[np.dtype(self.coordinates.dtype).char]
+            comm.Alltoallv([coords, rcount, rdisp, mpitype],
+                           [gathered, scount, sdisp, mpitype])
+
+            self._coordinates.data._local[:] = gathered[self._dist_subfunc_gather_mask]
 
         # Note: this method "mirrors" `_dist_scatter`: a sparse point that is sent
```

Comment:
Maybe rephrase? I am confused a bit here.

Reply:
I rather have another comment: @mloubout read this
"... However, the `coordinates` SubFunction values are not distributed, as this is a read-only field."

is this assumption what you're actually dropping in this PR?